src/ember-app.js

```diff
@@ -37,10 +37,15 @@ class EmberApp {
     this.vendorFilePaths = config.vendorFiles;
     this.moduleWhitelist = config.moduleWhitelist;
     this.hostWhitelist = config.hostWhitelist;
-    this.appConfig = config.appConfig;
+    this.config = config.config;
+    this.appName = config.appName;
 
     if (process.env.APP_CONFIG) {
-      this.appConfig = JSON.parse(process.env.APP_CONFIG);
+      let appConfig = JSON.parse(process.env.APP_CONFIG);
+      let appConfigKey = `${this.appName}/config/environment`;
+      if (!appConfig.hasOwnProperty(appConfigKey)) {
+        this.config[appConfigKey] = appConfig;
+      }
```

Comment:
We should probably create another environment variable to allow the entire config to be replaced, thoughts?

Reply:
What's the use case for that?

Reply:
Well, the same as the use case for `process.env.APP_CONFIG` in the first place.  To allow customizing the configuration without requiring a recompile.

Reply:
Oho yes, I agree. Will add that.